psydac/feec/multipatch/examples_nc/hcurl_eigen_pbms_nc.py

```diff
@@ -0,0 +1,379 @@
+        print('jump stabilization matrix...')
+        jump_stab_m = I1_m - cP1_m
+        JS_m = jump_stab_m.transpose() @ H1_m @ jump_stab_m
+
+    if generalized_pbm:
+        print('adding jump stabilization to RHS of generalized eigenproblem...')
+        B_m = cP1_m.transpose() @ H1_m @ cP1_m + JS_m
+    else:
+        B_m = H1_m
+
+    t_stamp = time_count(t_stamp)
+    print('solving matrix eigenproblem...')
+    all_eigenvalues, all_eigenvectors_transp = get_eigenvalues(
+        nb_eigs_solve, sigma, A_m, B_m)
+    # Eigenvalue processing
+    t_stamp = time_count(t_stamp)
+    print('sorting out eigenvalues...')
+    zero_eigenvalues = []
+    if skip_eigs_threshold is not None:
+        eigenvalues = []
+        eigenvectors = []
+        for val, vect in zip(all_eigenvalues, all_eigenvectors_transp.T):
+            if abs(val) < skip_eigs_threshold:
+                zero_eigenvalues.append(val)
+                # we skip the eigenvector
+            else:
+                eigenvalues.append(val)
+                eigenvectors.append(vect)
+    else:
+        eigenvalues = all_eigenvalues
+        eigenvectors = all_eigenvectors_transp.T
+
+    for k, val in enumerate(eigenvalues):
+        diags['eigenvalue_{}'.format(k)] = val  # eigenvalues[k]
+
+    for k, val in enumerate(zero_eigenvalues):
+        diags['skipped eigenvalue_{}'.format(k)] = val
+
+    t_stamp = time_count(t_stamp)
+    print('plotting the eigenmodes...')
+
+   # OM = OutputManager('spaces.yml', 'fields.h5')
+   # OM.add_spaces(V1h=V1h)
+
+    nb_eigs = len(eigenvalues)
+    for i in range(min(nb_eigs_plot, nb_eigs)):
+        OM = OutputManager(plot_dir + '/spaces.yml', plot_dir + '/fields.h5')
+        OM.add_spaces(V1h=V1h)
+        print('looking at emode i = {}... '.format(i))
+        lambda_i = eigenvalues[i]
+        emode_i = np.real(eigenvectors[i])
+        norm_emode_i = np.dot(emode_i, H1_m.dot(emode_i))
+        eh_c = emode_i / norm_emode_i
+        stencil_coeffs = array_to_psydac(cP1_m @ eh_c, V1h.vector_space)
+        vh = FemField(V1h, coeffs=stencil_coeffs)
+        OM.set_static()
+        # OM.add_snapshot(t=i , ts=0)
+        OM.export_fields(vh=vh)
+
+        # print('norm of computed eigenmode: ', norm_emode_i)
+        # plot the broken eigenmode:
+        OM.export_space_info()
+        OM.close()
+
+        PM = PostProcessManager(
+            domain=domain,
+            space_file=plot_dir +
+            '/spaces.yml',
+            fields_file=plot_dir +
+            '/fields.h5')
+        PM.export_to_vtk(
+            plot_dir +
+            "/eigen_{}".format(i),
+            grid=None,
+            npts_per_cell=[6] *
+            2,
+            snapshots='all',
+            fields='vh')
+        PM.close()
+
+        t_stamp = time_count(t_stamp)
+
+    return diags, eigenvalues
+
+
+def get_eigenvalues(nb_eigs, sigma, A_m, M_m):
+    """
+    Compute the eigenvalues of the matrix A close to sigma and right-hand-side M
```

Comment:
is this function specific to non-conforming grids ?

Reply:
No, it just tries to calculate eigenvalues of matrix equations. (As the doc string says) It's the same function as in `psydac/feec/multipatch/examples/hcurl_eigen_pbms_conga_2d.py`, we could also move this function to some utilities file.